solr/core/src/java/org/apache/solr/handler/component/HighlightComponent.java

```diff
@@ -189,7 +189,7 @@ protected SolrHighlighter getHighlighter(SolrParams params) {
 
     switch (method) {
       case UNIFIED:
-        if (solrConfigHighlighter instanceof UnifiedSolrHighlighter) {
+        if (solrConfigHighlighter.getClass() == UnifiedSolrHighlighter.class) {
```

Comment:
We must allow users to subclass UnifiedSolrHighlighter.  The solution to the conundrum here is to check that the class _also_  NOT a subclass of the PostingsSolrHighlighter.

Reply:
Ah, yes, you are indeed correct.